AsyncDisplayKit/ASMapNode.mm

```diff
@@ -207,7 +207,10 @@ - (void)takeSnapshot
                     [image drawAtPoint:CGPointZero];
                     
                     // Get a standard annotation view pin. Future implementations should use a custom annotation image property.
-                    MKAnnotationView *pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
+                    __block MKAnnotationView *pin;
+                    dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
It looks like we can check for this in setAnnotations:.  We probably want to create a +pinImage method, and in there, have a static UIImage *__pinImage = nil, then dispatch_once to get the image.  It should assert that inside the dispatch once is done on the main thread, but subsequent accesses could be off main if we can find a way to regulate that properly.

Seems doable, but definitely a) annoying to get right, and b) shitty that this code existed this way.  Thanks for pointing it out.